internals/api/dirfs.go

```diff
@@ -149,18 +149,18 @@ func (t Tree) AbsSecretPath(secretID *uuid.UUID) (*SecretPath, error) {
 // AbsDirPath returns the full path of dir
 // This function makes the assumption that only the root dir has no parentID.
 // If not, an error will occur.
-func (t Tree) AbsDirPath(dirID *uuid.UUID) (DirPath, error) {
-	if uuid.Equal(dirID, t.RootDir.DirID) {
+func (t Tree) AbsDirPath(dirID uuid.UUID) (DirPath, error) {
+	if uuid.Equal(&dirID, &t.RootDir.DirID) {
 		dirPath := t.ParentPath.JoinDir(t.RootDir.Name)
 		return dirPath, nil
 	}
 
-	dir, ok := t.Dirs[*dirID]
+	dir, ok := t.Dirs[dirID]
 	if !ok {
 		return "", ErrDirNotFound
 	}
 
-	parentPath, err := t.AbsDirPath(dir.ParentID)
+	parentPath, err := t.AbsDirPath(*dir.ParentID)
```

Comment:
This seems like we can avoid the cast here by looking at the signature of either the type or the function?

Reply:
What do you mean by that? `t.AbsDirPath` traverses the tree to get the full path of the given directory. It's not a type cast.

Reply:
Ah I said `cast` but I meant `pointer dereference`, brain fart I guess.

Reply:
The pointer dereference is made because the parent id field is optional. It is not set for root directories. As we've checked whether the directory is the root directory on the first line of this function, we can safely dereference here.